pyo3-macros-backend/src/pyclass.rs

```diff
@@ -416,41 +464,37 @@ pub fn build_py_enum(
     if enum_.variants.is_empty() {
         bail_spanned!(enum_.brace_token.span => "Empty enums can't be #[pyclass].");
     }
-    let variants: Vec<PyClassEnumVariant> = enum_
-        .variants
-        .iter()
-        .map(extract_variant_data)
-        .collect::<syn::Result<_>>()?;
-    impl_enum(enum_, args, variants, method_type, options)
-}
-
-fn impl_enum(
-    enum_: &syn::ItemEnum,
-    args: &PyClassArgs,
-    variants: Vec<PyClassEnumVariant>,
-    methods_type: PyClassMethodsType,
-    options: PyClassPyO3Options,
-) -> syn::Result<TokenStream> {
-    let enum_name = &enum_.ident;
     let doc = utils::get_doc(
         &enum_.attrs,
         options
             .text_signature
             .as_ref()
             .map(|attr| (get_class_python_name(&enum_.ident, args), attr)),
     );
+    let enum_ = PyClassEnum::new(enum_)?;
+    impl_enum(enum_, args, doc, method_type, options)
+}
```

Comment:
I know this isn't completely new, but why are those two functions split up? `impl_enum` is not called anywhere else and now it just adds a single line of code.

Reply:
It used to be more complex, then became one-liner after some refactoring. I did consider removing it, but didn't because I plan to expand `impl_enum` to implement #417. Maybe I should add some `// TODO` in the code?

Reply:
I think better to merge `impl_enum` and `impl_enum_class` for now than leave TODOs. Even if you get to #417 quickly it'll be trivial to split out again if that is what you need.